File: server/php/src/Presentation/Models/HostedCheckout/Response.php
<?php

namespace OnlinePayments\ExampleApp\Presentation\Models\HostedCheckout;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;
use OnlinePayments\ExampleApp\Application\Interfaces\Responses\ResponseInterface;

class Response implements ResponseInterface
{
    public string $hostedCheckoutId;
    public string $redirectUrl;
    public ?string $returnMAC;
    public string $amount;
    public ?Currency $currency;

    public function __construct(
        string $hostedCheckoutId,
        string $redirectUrl,
        ?string $returnMAC,
        string $amount,
        ?Currency $currency
    ) {
        $this->hostedCheckoutId = $hostedCheckoutId;
        $this->redirectUrl = $redirectUrl;
        $this->returnMAC = $returnMAC;
        $this->amount = $amount;
        $this->currency = $currency;
    }

    public function toArray(): array
    {
        return [
            'hostedCheckoutId' => $this->hostedCheckoutId,
            'redirectUrl' => $this->redirectUrl,
            'returnMAC' => $this->returnMAC,
            'amount' => $this->amount,
            'currency' => $this->currency,
        ];
    }
}

File: server/ruby/app/presentation/models/create_hosted_checkout/response.rb
module Presentation
  module Models
    module CreateHostedCheckout
      class Response
        attr_reader :hosted_checkout_id, :redirect_url, :return_mac, :amount, :currency

        def initialize(hosted_checkout_id:, redirect_url:, return_mac:, amount:, currency:)
          @hosted_checkout_id = hosted_checkout_id
          @redirect_url      = redirect_url
          @return_mac        = return_mac
          @amount            = amount
          @currency          = currency
        end

        def as_json(*)
          {
            hostedCheckoutId: hosted_checkout_id,
            redirectUrl: redirect_url,
            returnMAC: return_mac,
            amount: amount,
            currency: currency
          }
        end
      end
    end
  end
end